test/functional/feature_staking_free_tx_preference.py

```diff
@@ -0,0 +1,103 @@
+from test_framework.messages import COIN
+from test_framework.staking_utils import FreeTransactionMixin
+from test_framework.test_framework import BitcoinTestFramework
+
+from time import sleep
+
+class FreeTxPreferenceTest(BitcoinTestFramework, FreeTransactionMixin):
+    def set_test_params(self):
+        self.blockmaxweight = 128000
+        self.num_nodes = 2
+        self.extra_args = [
+            ["-txindex", f'-blockmaxweight={self.blockmaxweight}'],
+            ["-txindex", f'-blockmaxweight={self.blockmaxweight}']
+        ]
+
+    def get_staking_pool_balance(self, node_num: int) -> int:
+        return self.nodes[node_num].getstakinginfo()['staking_pool']
+
+    def send_staking_deposit_tx(self, stake_address: str, deposit_value: float, node_num: int):
+        txid = self.nodes[node_num].depositstake(deposit_value, 4320, stake_address)
+        return txid
+
+    def run_test(self):
+        self.free_tx_trasaction_preference()
+
+    def assert_block_tx_preference(self, free_tx_list):
+        self.nodes[0].generate(1)
+        last_block = self.nodes[0].getblock(self.nodes[0].getbestblockhash())
+
+        free_tx_size = 0
+        nonfree_tx_size = 0
+        for tx in last_block["tx"]:
+            tx_size = len(
+                bytes.fromhex(
+                    self.nodes[0].gettransaction(tx)['hex']
+                    )
+                )
+
+            if tx in free_tx_list:
+                free_tx_size = free_tx_size + tx_size
+            else:
+                nonfree_tx_size = nonfree_tx_size + tx_size
+
+        free_tx_percentage = free_tx_size / self.blockmaxweight
+        assert free_tx_percentage < .28 and free_tx_percentage > .24 # TODO figure out how to make it more precise
```

Comment:
The desired block size should be defined as a constant at the beginning of the test and should correspond to the parameter in chainparams. 
Also, consensus has the max free tx size defined as an absolute value, not as the percentage of the block size. If we use the lowered blockweight we're in fact skipping the consensus validation of the mining algorithm, which is a very important case. I would try testing on a full block size.
Also, if the upper limit has to be as big as .28, the current algorithm is not working properly - it shouldn't pass any block with more than 25% of free tx size.